base/pmap.jl

```diff
@@ -15,38 +19,170 @@ Note that `f` must be made available to all worker processes; see
 and Loading Packages <man-parallel-computing-code-availability>`)
 for details.
 """
-function pgenerate(p::WorkerPool, f, c)
-    if length(p) == 0
-        return AsyncGenerator(f, c)
+function pgenerate(p::WorkerPool, f, c; distributed=true, batch_size=1, on_error=nothing,
+                                        retry_n=0,
+                                        retry_max_delay=DEFAULT_RETRY_MAX_DELAY,
+                                        retry_on=DEFAULT_RETRY_ON)
+    # Don't do remote calls if there are no workers.
+    if (length(p) == 0) || (length(p) == 1 && fetch(p.channel) == myid())
+        distributed = false
+    end
+
+    # Don't do batching if not doing remote calls.
+    if !distributed
+        batch_size = 1
+    end
+
+    # If not batching, do simple remote call.
+    if batch_size == 1
+        if distributed
+            f = remote(p, f)
+        end
+
+        if retry_n > 0
+            f = wrap_retry(f, retry_on, retry_n, retry_max_delay)
```

Comment:
This results in a stack overflow
```
f = () -> 1
f = () -> f() + 1
f()
```

This works
```
f = ()->1
function wrap(f)
  () -> f() + 1
end
f = wrap(f)
f()
```

I don't know if this is by design. Should open an issue if not. 

Different variable names for `f` at each step would have removed the need for the wrapping function.